tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11135,402 @@ TEST_F(ExecutionTest, PackUnpackTest) {
   }
 }
 
+// TODOLongVec : Need to change the members to vectors. But when I did that the
+// copy logic to read back from the shader buffer fails. Needs to fix that
+// before checking in PR.
+// SLongVectorBinaryOp is used in ShaderOpArithTable.xml. The shader program
+// uses the struct defintion to read from the input global buffer.
+template <typename T, std::size_t N = 4,
+typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+struct SLongVectorBinaryOp {
+  T scalarInput;
+  std::array<T, N> vecInput1;
+  std::array<T, N> vecInput2;
+  std::array<T, N> vecOutput;
+};
+
+// vec1 == Expected vector
+// vec2 == Actual vector
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
```

Comment:
Whoops, I intended to either add an index parameter to this function or have the callers also log the index of the array they are testing. They're arrays of arrays.